### PR TITLE
Use debug from npm instead of custom debug solutions

### PR DESCRIPTION
**Pull request by j-f1**
This uses npm’s [`debug`](https://www.npmjs.com/package/debug) module to manage logging.

### CODE REVIEW

**Comment by j-f1** [comment]
@ArtOfCode- Why?

**Comment by ArtOfCode-** [comment]
@j-f1 Mostly because in all likelihood I did it like that for a reason

**Comment by ArtOfCode-** [comment]
^ FDSC approved, other script authors should look at their own stuff

File: fdsc/fdsc.user.js [hunk 1]
```diff
@@ -23,14 +24,17 @@
...
 // @grant       none
 // ==/UserScript==
 
-/* global fdsc, $, xdLocalStorage, confirm */
+/* global fdsc, $, xdLocalStorage, confirm, unsafeWindow */
 /* eslint-disable max-nested-callbacks */
 
 (function () {
   "use strict";
+  const createDebug = typeof unsafeWindow === "undefined" ? window.debug : unsafeWindow.debug || window.debug;
```
Comment (Cerbrus) [inline fdsc/fdsc.user.js]: Wouldn't this be functionally equal, but cleaner?

    const createDebug = (unsafeWindow || window).debug;

Comment (j-f1) [inline fdsc/fdsc.user.js]: @Cerbrus If `unsafeWindow` is undefined, that’ll throw a `ReferenceError`.

**Comment by j-f1** [comment]
@Ferrybig Does Spamtracker work again?

**Comment by ferrybig** [comment]
@j-f1 No, Still the same error.

> @ jf After a look through the code, it seems UnsafeWindow is defined as window using the IIFE, for some reason your script is probably included in the global window scope, we can add another argument for this "UserScriptManagerWindow" argument

https://chat.stackexchange.com/transcript/message/37405109#37405109

File: fdsc/fdsc.user.js (hunk 1 above)
Comment (Cerbrus) [inline fdsc/fdsc.user.js]: Aw shoot, you're right @j-f1. That's a shame.

**Comment by j-f1** [comment]
@ferrybig Are you using Greasemonkey?

**Comment by ferrybig** [comment]
@j-f1 tamper monkey on Google chrome.

Could it be the case that it works for you because another userscript is importing the debug library in the window variable, as haven't installed the other userscripts

File: spamtracker/spamtracker.user.js [hunk 2]
```diff
@@ -25,9 +25,9 @@
 /* global GM_info, Notification, GM_setValue, GM_getValue, unsafeWindow, GM_getResourceText, GM_getResourceURL */
 /* eslint-disable prefer-const, no-use-before-define */
 
-unsafeWindow.Spamtracker = (function (target, siterooms, window) {
+unsafeWindow.Spamtracker = (function (target, siterooms, window, orginalWindow) {
   "use strict";
-  const createDebug = typeof unsafeWindow === "undefined" ? window.debug : unsafeWindow.debug || window.debug;
+  const createDebug = typeof orginalWindow === "undefined" ? window.debug : orginalWindow.debug || window.debug;
```
Comment (ArtOfCode-) [inline spamtracker/spamtracker.user.js]: s/orginal/original/g

**Pull request merged**